server/apps/rules/routing_rules.py

```diff
@@ -21,6 +21,15 @@
 STATE_ROUTED = 'routed'
 
 
+WEEKDAYS = ['MON', 'TUE', 'WED', 'THU', 'FRI', 'SAT', 'SUN']
+
+
+def parse_time(current_datetime, timestr=None):
+    if timestr is None:
+        timestr = '0000'
+    return current_datetime.replace(hour=int(timestr[:2]), minute=int(timestr[-2:]), second=0)
```

Comment:
Would it be better to use `datetime.strptime` for parsing the string? It would, among other things, catch a few more potential format errors of the `timestr`. For example, passing in a time also containing the seconds part, e.g. `22:04:17` would slip through and be recognized as `22:17:00`.

A docstring would be highly useful as well, describing what an expected input is.